backend/tests/integration/pidstore/minters/test_minters_texkey.py

```diff
@@ -0,0 +1,216 @@
+        PersistentIdentifier.pid_type == "texkey"
+    ).all()
+    assert len(pids) == 2
+    assert pids[0].status == PIDStatus.DELETED
+    assert pids[1].status == PIDStatus.DELETED
+
+
+def test_minter_texkey_featureflag_test(inspire_app):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=False):
+        record = create_record("lit", data=data)
+
+    pid_count = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .count()
+    )
+    assert pid_count == 0
+
+
+def test_minter_ignores_texkey_pid_when_missing_data_to_create_it(inspire_app):
+    data = {"authors": [{"full_name": "Janeway, K."}]}
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+    pid_count = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .count()
+    )
+
+    assert "texkey" not in record
+    assert pid_count == 0
+
+    data = {"publication_info": [{"year": 2000}]}
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+    pid_count = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .count()
+    )
+
+    assert "texkey" not in record
+    assert pid_count == 0
+
+
+def test_minter_creates_record_with_texkey_provided_and_new_one_when_metadata_requires_it(
+    inspire_app
+):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+        "texkeys": ["Janeway:1999abc", "Janewaj:2000def"],
+    }
+
+    expected_existing_texkeys = ["Janeway:1999abc", "Janewaj:2000def"]
+    expected_new_texkey = "Janeway:2000"
+    expected_texkeys_count = 3
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+
+    assert len(record["texkeys"]) == expected_texkeys_count
+
+    pids_from_pidstore = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .all()
+    )
+
+    assert len(pids_from_pidstore) == expected_texkeys_count
+    assert record["texkeys"][0].startswith(expected_new_texkey)
+    assert record["texkeys"][1] in expected_existing_texkeys
+    assert record["texkeys"][2] in expected_existing_texkeys
+
+    all_expected_texkeys = record["texkeys"]
+
+    assert pids_from_pidstore[0].pid_value in all_expected_texkeys
+    assert pids_from_pidstore[1].pid_value in all_expected_texkeys
+    assert pids_from_pidstore[2].pid_value in all_expected_texkeys
```

Comment:
what about test cases for:
* undeleting a record
* stealing a texkey from a deleted record

Reply:
:+1:  Currently we are not supporting un-deleteing on control_numbers (I'm not sure about other PIDs) but for texkey it will work. (I will add test)